File: Src/PRuntimes/PSymRuntime/src/main/java/psym/runtime/values/PMachineValue.java
package psym.runtime.values;

import psym.runtime.machine.Machine;

public class PMachineValue extends PValue<PMachineValue> {
  // stores the int value
  private final Machine value;

  public PMachineValue(Machine val) {
    value = val;
  }

  public PMachineValue(Object val) {
    if (val instanceof PMachineValue) value = ((PMachineValue) val).value;
    else value = (Machine) val;
  }

  public PMachineValue(PMachineValue val) {
    value = val.value;
  }

  public Machine getValue() {
    return value;
  }

  public int getId() {
    return value.getInstanceId();
  }

  @Override
  public PMachineValue clone() {
    return new PMachineValue(value);
  }

  @Override
  public int hashCode() {
    return value.hashCode();
  }

  @Override
  public boolean equals(Object obj) {
    if (obj == this) return true;
    else if (!(obj instanceof PMachineValue)) {
      return false;
    }
    return this.value.equals(((PMachineValue) obj).value);
  }

  @Override
  public String toString() {
    return value.toString();
  }
}

File: Src/PChecker/CheckerCore/Runtime/Values/PMachineValue.cs
using System.Collections.Generic;
using System.Linq;
using PChecker.Runtime.StateMachines;

namespace PChecker.Runtime.Values
{
    public class I_Main : PMachineValue
    {
        public I_Main(StateMachineId machine, List<string> permissions) : base(machine, permissions)
        {
        }
    }

    public class PMachineValue : IPValue
    {
        public PMachineValue(StateMachineId machine, List<string> permissions)
        {
            Id = machine;
            Permissions = permissions.ToList();
        }

        public PMachineValue(PMachineValue mValue)
        {
            Id = mValue.Id;
            Permissions = mValue.Permissions.ToList();
        }

        public StateMachineId Id { get; }
        public List<string> Permissions { get; }

        public bool Equals(IPValue other)
        {
            return other is PMachineValue machine && Equals(Id, machine.Id);
        }

        public override int GetHashCode()
        {
            return Id.GetHashCode();
        }

        public IPValue Clone()
        {
            return new PMachineValue(Id, new List<string>(Permissions));
        }

        public override string ToString()
        {
            return Id.Name.Split('.').Last();
        }

        public object ToDict()
        {
            return ToString();
        }
    }
}